### DISCUSSION TITLE
Hybrid App, published, PII - can't create restrictedDataToken in Postman, not even in sandbox

### DISCUSSION THREAD

**Discussion by Landsailor** [amzn/selling-partner-api-models#3062] [General] [answered]
I've already opened a support request, so I'm looking for any clues to help, well, get the support process moving again after being stalled for over a week.

We used MWS to process orders for numerous client for _years_. Earlier this year, I created the Hybrid app following the example in the documentation, updated the system, got the "app" approved and published. We're approved for Direct to Consumer shipping, so we're good there.

We converted from MWS to SP-API, but did _not_ have an MWS App, per se, listed in our Seller Account to convert. Our clients generated the needed MWS Auth Tokens manually by entering our developer name and ID in the "Authorize New Developer" tab from "Manage Your Apps".

Once published, I was able to use the Authorization API to easily convert their MWS credentials. Everything went smoothly. I just had a client renew his authorization this morning. All is well....except...

**Generate Restricted Data Token fails, even in the sandbox**
- Postman - responds with a 403 - "Access to requested resource is denied."
- Server - the scripts _will_ generate an RDT using the correct endpoint, but it's useless (Hoping that a full SP-API authorization would make a difference, I just confirmed with an RDT from the client who renewed today. Still no good.)

Everything else is working just fine. I can call get Order Address just fine with the refresh token, but it fails with the RDT.

The Role ARN for the app was created correctly, AFAIK.

Does anyone have any ideas?

**Comment by Landsailor** [comment, marked as answer]
Always be sure your URLs are correct, kids! Look for any trailing characters due to a copy and paste error. That resolves the Postman issue. However, when the Amazon documentation is wrong, they're just setting people up to fail.

If your RDT won't actually retrieve any Order data and you've followed the [Tutorial: Get authorization to access PII for bulk orders](https://developer-docs.amazon.com/sp-api/docs/tokens-api-use-case-guide#step-1-get-an-rdt), allow me to save you possibly _hours and hours_ of troubleshooting work: Be _ sure your body _includes_ "{orderId}"!

### Correct Payload to access PII for Bulk Orders
`{
	"restrictedResources": [{
		"method": "GET",
		"path": "/orders/v0/orders/{orderId}",
		"dataElements": ["buyerInfo", "shippingAddress"]
	}]
}`
<img width="546" alt="Screen Shot 2022-09-20 at 2 21 19 PM" src="https://user-images.githubusercontent.com/1393119/191366902-515de242-01df-4870-8e85-3c25ca09f2db.png">


As of September 20, 2022, several weeks after I submitted the documentation error to the Amazon rep who was trying to help, it is _still_ showing this incorrect example:
![Screen Shot 2022-09-20 at 2 19 53 PM](https://user-images.githubusercontent.com/1393119/191366732-07baf3b7-6588-4a4e-a6e2-22181ce6c928.png)

[Further down the page ](https://developer-docs.amazon.com/sp-api/docs/tokens-api-use-case-guide#terminology) in the "Terminology" section, it states:
 **Generic path**. A path in a restricted resource that contains a generic identifier, **_such as {orderId}_** or {shipmentId}. For example, orders/v0/orders/{orderId}/address

Add the generic identifier to Amazon's incorrect example and it will work.

**Reply by vaidas-lungis** [reply]
OMG.. 😰 🥵
